### ISSUE TITLE
Getting a `ParseError` rather than an `HTTPError`

### ISSUE THREAD

**Issue by StevenXL** [phadej/github#353]
Hello. I tried to upgrade the `github` package from `v0.20` to `v0.21` in a project. When I did so, the behavior of the package changed.

On `v0.20`, if I tried to edit an issue (a `patch` request) that does not exist, then I would receive a value of type `Error` tagged with `HTTPError`. On `v0.21`, I instead get a value of type `Error` tagged with `ParseError`.  (An example of the `HTTPError` and `ParseError` are below). 

I believe that the library is **not** recognizing 404 errors being returned from the server, and instead is trying to parse the the body `{
    "message": "Not Found",
    "documentation_url": "https://developer.github.com/v3/issues/#get-a-single-issue"
}` into a value of type `Issue`.

```
  HTTPError (HttpExceptionRequest Request {
  host                 = "api.github.com"
  port                 = 443
  secure               = True
  requestHeaders       = [("Authorization","<REDACTED>"),("User-Agent","github.hs/0.7.4"),("Accept","application/vnd.github.preview")]
  path                 = "/repos/StevenXL/hit-on/issues/99"
  queryString          = ""
  method               = "GET"
  proxy                = Nothing
  rawBody              = False
  redirectCount        = 10
  responseTimeout      = ResponseTimeoutDefault
  requestVersion       = HTTP/1.1
}
 (StatusCodeException (Response {responseStatus = Status {statusCode = 404, statusMessage = "Not Found"}, responseVersion = HTTP/1.1, responseHeaders = [("Server","GitHub.com"),("Date","Fri, 22 Mar 2019 18:09:29 GMT"),("Content-Type","application/json; charset=utf-8"),("Transfer-Encoding","chunked"),("Status","404 Not Found"),("X-RateLimit-Limit","5000"),("X-RateLimit-Remaining","4999"),("X-RateLimit-Reset","1553281769"),("X-OAuth-Scopes","admin:business, admin:gpg_key, admin:org, admin:org_hook, admin:public_key, admin:repo_hook, delete_repo, gist, notifications, repo, user, write:discussion"),("X-Accepted-OAuth-Scopes","repo"),("X-GitHub-Media-Type","github.v3; param=preview"),("Access-Control-Expose-Headers","ETag, Link, Location, Retry-After, X-GitHub-OTP, X-RateLimit-Limit, X-RateLimit-Remaining, X-RateLimit-Reset, X-OAuth-Scopes, X-Accepted-OAuth-Scopes, X-Poll-Interval, X-GitHub-Media-Type"),("Access-Control-Allow-Origin","*"),("Strict-Transport-Security","max-age=31536000; includeSubdomains; preload"),("X-Frame-Options","deny"),("X-Content-Type-Options","nosniff"),("X-XSS-Protection","1; mode=block"),("Referrer-Policy","origin-when-cross-origin, strict-origin-when-cross-origin"),("Content-Security-Policy","default-src 'none'"),("Content-Encoding","gzip"),("X-GitHub-Request-Id","F792:27E8:38E401:9238DE:5C9524D9")], responseBody = (), responseCookieJar = CJ {expose = []}, responseClose' = ResponseClose}) "{\"message\":\"Not Found\",\"documentation_url\":\"https://developer.github.com/v3/issues/#get-a-single-issue\"}"))
```

```
ParseError "Error in $: key \"updated_at\" not present"
```

**Comment by phadej** [comment]
It's probably the same as #352; i.e. I broke error status check. Unfortunately I paged out what I did, so I cannot say how easy or hard is to fix that.

**Comment by StevenXL** [comment]
@phadej I'll take a look. Frankly, I'm only just now getting into type-level progamming in Haskell, so I am not sure how much help I can be. (Although I am having a blast spelunking through the code and actually being able to follow what is going on with `GenRequest`, etc). 

If I have anything of value to add, I'll post again, but again, newbie here.

**Comment by phadej** [comment]
I think `makeHttpRequest` https://github.com/phadej/github/blob/0c31bbd7d2ccca60171ba8be59082031e98a749d/src/GitHub/Request.hs#L310-L340 should install [`checkResponse`](http://hackage.haskell.org/package/http-client-0.6.2/docs/Network-HTTP-Client.html#v:checkResponse) based on `Accept mt`

So I think: https://github.com/phadej/github/blob/0c31bbd7d2ccca60171ba8be59082031e98a749d/src/GitHub/Request.hs#L179 is incorrect as we'd should check for `200` status by default. And currently we do nothing.

**Comment by StevenXL** [comment]
The lack of `checkResponse` is indeed where I ended up. I didn't even know that the `http-client` library allowed for that. (Definitely a feature I'll be using more often). 

Thanks for the info - going to start looking at `modifyRequest`.

**Comment by phadej** [comment]
Fixed. Will be released in in `0.22`
